gk/fib.c

```diff
@@ -1664,3 +1665,303 @@ del_fib_entry(const char *ip_prefix, struct gk_config *gk_conf)
 
 	return del_fib_entry_numerical(&prefix_info, gk_conf);
 }
+
+struct gk_fib_ether_entry {
+
+	bool          stale;
+
+	/* The IP address of the nexthop. */
+	struct ipaddr nexthop_ip;
+
+	uint32_t      ref_cnt;
+
+	/* The whole link-layer header. */
+	struct {
+		/* Ethernet header (required). */
+		struct ether_hdr eth_hdr;
+		/* VLAN header (optional). */
+		struct vlan_hdr  vlan_hdr;
+	} __attribute__((packed)) l2_hdr;
+};
+
+struct gk_fib_dump_entry {
+
+	/* The IP prefix. */
+	struct ipaddr addr;
+
+	int           len;
+
+	/* The Grantor IP address. */
+	struct ipaddr grantor_ip;
+
+	uint16_t      num_ether_entries;
```

Comment:
Isn't `num_ether_entries` always `1`? When do you see `num_ether_entries` not being `1`? If it's always `1`, we can simplify the code.

Reply:
No, it's not always the case. When we are dealing with the [(neighbor table),](https://github.com/AltraMayor/gatekeeper/pull/89/files#diff-dc647b4a85684f2cb1539dac40550b60R1788) we need to dump all the entries in that table,

Reply:
Thanks. I had missed that case.